### PR TITLE
Add Configurable usage agreement

### PR DESCRIPTION
**Pull request by MoritzHahn1337**

### CODE REVIEW

File: datameta/scripts/initialize_db.py [hunk 1]
```diff
@@ -207,6 +207,10 @@ def create_default_site_settings(db):
         key='logo_html',
         str_value = '<p class="h4 my-0 me-md-auto fw-normal" style="font-family: \'Fira Sans\', sans-serif;"><a href="/" class="link-bare"><span style="color:#ffca2c">D</span>ata<span style="color:#ffca2c">M</span>eta</a></p>'
         ))
+    db.add(ApplicationSettings(
+        key='usage_agreement',
```
Comment (lkuchenb) [inline datameta/scripts/initialize_db.py:211]:
```suggestion
        key='user_agreement',
```

**Review by lkuchenb — CHANGES_REQUESTED** [review]

**Comment by lkuchenb** [comment]
Please rebase, I had to pull in the `site_read` fix in #358

**Review by lkuchenb — CHANGES_REQUESTED** [review]
Thanks for the refactoring! See inline comments and the following:

The error response should be a 401 now. Please remove the `RegistrationResponse` and turn that into a `204`. For errors, use `errors.get_validation_error` as usual. The validation error schema has a `fields` field which we could use for encoding the affected fields, each one needs a message though which could be empty in our case:
```python
errors.get_validation_error(fields=error_fields, messages=['' for _ in error_fields])
```
where `error_fields` is just a list of failed fields (we don't need the bool actually).

**Comment by lkuchenb** [comment]
@MoritzHahn1337 Please rebase and re-bump the openapi version, I just merged a bump into develop

**Comment by MoritzHahn1337** [comment]
> `RegistrationResponse`

Do you mean 400 instead of 401?

**Comment by MoritzHahn1337** [comment]
> 
> 
> Thanks for the refactoring! See inline comments and the following:
> 
> The error response should be a 401 now. Please remove the `RegistrationResponse` and turn that into a `204`. For errors, use `errors.get_validation_error` as usual. The validation error schema has a `fields` field which we could use for encoding the affected fields, each one needs a message though which could be empty in our case:
> 
> ```python
> errors.get_validation_error(fields=error_fields, messages=['' for _ in error_fields])
> ```
> 
> where `error_fields` is just a list of failed fields (we don't need the bool actually).

@lkuchenb do you meen 400 isntead of 401?

**Comment by lkuchenb** [comment]
Yep, sorry, 400

File: datameta/static/js/register.js [hunk 2]
```diff
@@ -24,47 +78,69 @@ window.addEventListener("load", function() {
...
+                elem_alert.style.display="none"
+                document.getElementById("regform").style.display="none"
+                document.getElementById("success").style.display="block"
+                return;
+            } else if(response.status === 400) {
+                response.json().then((json) => {
+                    var errorFields = [];
+                    json.forEach(function (error) {
+                        errorFields.push(error.field)
+                    })
```
Comment (lkuchenb) [inline datameta/static/js/register.js:111]:
```suggestion
                    var errorFields = json.map(error => error.field);
```

File: datameta/templates/register.pt [hunk 3]
```diff
@@ -56,7 +56,7 @@
 </style>
 <main metal:fill-slot="content" class="form-signin">
         <div tal:replace="structure logo_html"></div>
-	<div id="alert" class="alert alert-danger mt-2" role="alert" style="display:none">
+	<div id="alert" class="alert alert-danger mt-2 alert-dismissible" role="alert" style="display:none">
 		A simple danger alert—check it out!
```
Comment (lkuchenb) [inline datameta/templates/register.pt:60]:
```suggestion
```

**Review by lkuchenb — APPROVED** [review]
Thanks, just some minor change suggestions below, feel free to merge after fixing

**Pull request merged**